### PR TITLE
Add RISC-V support

### PR DESCRIPTION
**Pull request by ethanhs**
Hopefully this is correct, do let me know if I've made a mistake, or you feel like something should be renamed.

### CODE REVIEW

**Comment by ethanhs** [comment]
I should note RISC-V is a modular ISA. 

As is, people need to specify e.g. `-mriscv=RV64IAMFD` (You may want to read https://en.wikipedia.org/wiki/RISC-V#ISA_Base_and_Extensions).

I see three options:
- require people to specify which extensions they want (I fear people would be suprised at the lack of mul by default)
- provide RV64G configurations
- offer both?

Thoughts/suggestions appreciated.

**Comment by mattgodbolt** [comment]
Hi @ethanhs - thanks for the PR!

Firstly, I tried this with the trunk clang (built with your patch, as best I can tell), and I still get an error:

```
ubuntu@admin-node /tmp> cat > test.cc
// Type your code here, or load an example.
int square(int num) {
    return num * num;
}

/opt/compiler-explorer/clang-trunk/bin/clang++ -S -c -o /tmp/moo.asm -target riscv64 test.cc
error: unable to create target: 'No available targets are compatible with this triple.'
1 error generated.
/opt/compiler-explorer/clang-trunk/bin/clang++ --version
clang version 7.0.0 (trunk 333657)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/compiler-explorer/clang-trunk/bin
```

Secondly, if we can make the regular clang compiler work with just adding `-target riscv64` then that's probably the right way to go (for now!). We have a whole bunch of targets that clang supports that don't have their own "target". Though I'm not against it, it would mean duplicating all the clangs for each supported target, for each supported clang version too (imagine how many more options that would add to the already overrun compiler dropdown!)

I'm not saying that's not the right thing to do: but we probably need a more general solution for clang's targets. According to https://stackoverflow.com/questions/15036909/clang-how-to-list-supported-target-architectures there's not a way to list them from the binary :/. The best I get is:

```
ubuntu@admin-node /tmp> /opt/compiler-explorer/clang-trunk/bin/llc --version
LLVM (http://llvm.org/):
  LLVM version 7.0.0svn
  Optimized build.
  Default target: x86_64-unknown-linux-gnu
  Host CPU: haswell

  Registered Targets:
    aarch64    - AArch64 (little endian)
    aarch64_be - AArch64 (big endian)
    amdgcn     - AMD GCN GPUs
    arm        - ARM
    arm64      - ARM64 (little endian)
    armeb      - ARM (big endian)
    bpf        - BPF (host endian)
    bpfeb      - BPF (big endian)
    bpfel      - BPF (little endian)
    hexagon    - Hexagon
    lanai      - Lanai
    mips       - Mips
    mips64     - Mips64 [experimental]
    mips64el   - Mips64el [experimental]
    mipsel     - Mipsel
    msp430     - MSP430 [experimental]
    nvptx      - NVIDIA PTX 32-bit
    nvptx64    - NVIDIA PTX 64-bit
    ppc32      - PowerPC 32
    ppc64      - PowerPC 64
    ppc64le    - PowerPC 64 LE
    r600       - AMD GPUs HD2XXX-HD6XXX
    sparc      - Sparc
    sparcel    - Sparc LE
    sparcv9    - Sparc V9
    systemz    - SystemZ
    thumb      - Thumb
    thumbeb    - Thumb (big endian)
    x86        - 32-bit X86: Pentium-Pro and above
    x86-64     - 64-bit X86: EM64T and AMD64
    xcore      - XCore
```

which seems to suggest the clang build didn't work after all... I'll check on that!

**Comment by mattgodbolt** [comment]
Right, seems the clang build isn't working. Investigating...

**Comment by mattgodbolt** [comment]
Hah, getting: 

```
svn: E000110: Error running context: Connection timed out
```

during update...wonder what's going on there...digging.

**Comment by mattgodbolt** [comment]
Ok temporary network problem...but:

```
build.sh: line 58: -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=RISCV: command not found
```

is the real error! hahah fixing :)

**Comment by ethanhs** [comment]
>Secondly, if we can make the regular clang compiler work with just adding -target riscv64 then that's probably the right way to go (for now!).

Okay, sounds good. And thank you again for fixing my mistake!

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -148,15 +148,12 @@ compiler.clang_concepts.options=-std=c++2a -Xclang -fconcepts-ts -stdlib=libc++
 compiler.clang_concepts.notification=Highly experimental compiler. Bug reports welcomed at <a href="https://github.com/saarraz/clang-concepts/issues" target="_blank" rel="noopener noreferrer">github.com/saarraz/clang-concepts/issues <sup><small class="glyphicon glyphicon-new-window opens-new-window" title="Opens in a new window"></small></sup></a>
 
 # Clang for RISC-V
-group.rvclang.compilers=rv32clang:rv64clang
+group.rvclang.compilers=rv32clang
 group.rvclang.groupName=Clang RISC-V
 group.rvclang.supportsBinary=false
 compiler.rv32clang.exe=/opt/compiler-explorer/clang-trunk/bin/clang++
-compiler.rv32clang.name=Clang RISC-V (32 bit)
+compiler.rv32clang.name=Clang riscv32 (trunk)
```
Comment (mattgodbolt) [inline etc/config/c++.amazon.properties]: One last nitpick: all the other architectures are listed first in their compiler name ("ARM gcc" ... "x86 clang" ... "MSP430 gcc" etc) ... although their group name is "Compiler Arch".

Can you change the name here to be "RISC-V clang (trunk)" (assuming the capitalisation is correct) and then I'll merge. We need a better solution to this in general. the dropdown is terrible already (see #137 #257 etc), but at least consistency with the (bad) current setup is better than none.

Comment (mattgodbolt) [reply etc/config/c++.amazon.properties]: Argh, it would need the 32-bit indicator too I guess, "riscv32 clang (trunk)" maybe?

**Comment by mattgodbolt** [comment]
Merged: will do a site update once things are built.

**Comment by ethanhs** [comment]
Thank you! I'm excited :)

**Pull request merged**